src/main/java/uk/gov/hmcts/reform/notificationservice/service/NotificationMessageParser.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.hmcts.reform.notificationservice.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.microsoft.azure.servicebus.MessageBody;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+import uk.gov.hmcts.reform.notificationservice.exception.InvalidMessageException;
+import uk.gov.hmcts.reform.notificationservice.model.request.incomming.NotificationMsg;
+
+import java.io.IOException;
+import java.util.List;
+
+@Service
+public class NotificationMessageParser {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NotificationMessageParser.class);
+    private static final String ERROR_CAUSE = "Message Binary data is null";
+
+    private final ObjectMapper objectMapper;
+
+    public NotificationMessageParser(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+    }
+
+    public NotificationMsg parse(MessageBody messageBody) {
+        try {
+            NotificationMsg notificationMsg =
+                    objectMapper.readValue(getBinaryData(messageBody), NotificationMsg.class);
+            LOGGER.info(
+                    "Parsed notification message, Zip File Name: {}, Error Code: {}, Error Description: {} "
+                            + "Jurisdiction: {}, PO Box: {}, Service: {}, Document Control Number: {}",
+                    notificationMsg.zipFileName,
+                    notificationMsg.errorCode,
+                    notificationMsg.errorDescription,
+                    notificationMsg.jurisdiction,
+                    notificationMsg.poBox,
+                    notificationMsg.service,
+                    notificationMsg.documentControlNumber
+            );
+
+            return notificationMsg;
+        } catch (IOException exc) {
+            LOGGER.error("Notification queue message parse error:", exc);
+            throw new InvalidMessageException(exc);
+        }
+    }
+
+    private static byte[] getBinaryData(MessageBody messageBody) {
+        List<byte[]> binaryData = messageBody.getBinaryData();
+        if (binaryData == null) {
+            throw new InvalidMessageException(ERROR_CAUSE);
```

Comment:
Again, could use message ID for back linking/tracking

Reply:
message has not been parsed in this point.

Reply:
message id is part of service bus message and not the body. should use that up until we know the ID of object for back tracing

Reply:
this class gets get messageBody, not the message.

Reply:
I think it's a good point. If we encounter an invalid message, it's important that we can identify it and track it. Maybe we should pass the whole message for that reason.

Reply:
On a second thought, the calling class will have to catch InvalidMessageException specifically. It can log message ID instead.

Reply:
pass the whole message

Reply:
then we can remove logging completely :+1: 

> On a second thought, the calling class will have to catch InvalidMessageException specifically. It can log message ID instead.

Reply:
it is a runtimeexception , I think we should log sth.

Reply:
removed